src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1399,24 +1401,26 @@ protected void ExecuteIndexing()
                         while (true)
                         {
 
-                            WaitHandle.WaitAny(new [] {_mre.WaitHandle, _rollingEvent.WaitHandle, _indexingProcessCancellationTokenSource.Token.WaitHandle});
+                            var result = WaitHandle.WaitAny(new [] {_mre.WaitHandle, _rollingEvent.WaitHandle, _indexingProcessCancellationTokenSource.Token.WaitHandle});
                             _indexingProcessCancellationTokenSource.Token.ThrowIfCancellationRequested();
                             
                             if (_indexDisabled)
                                 return;
-
-                            _rollingEvent.Reset();
+                            
+                            _mre.Reset();
 
                             var replaceStatus = ReplaceIfNeeded(batchCompleted: false, didWork: false);
 
                             if (replaceStatus == ReplaceStatus.Succeeded)
                                 return;
 
                             if (replaceStatus == ReplaceStatus.NotNeeded)
-                                break;
+                            {
+                                if (result != 0)
```

Comment:
I wonder if wouldn't be more clear and explicit to check that `_mre` isn't set, instead of relaying on an index.

Reply:
sure, I see that no-one calls `_mre.Reset()` from the outside so it should be safe.